packages/ruleset/src/collections/index.js

```diff
@@ -0,0 +1,19 @@
+// a group of predefined "collections" of OpenAPI locations to validate
+// helpful when the same group of locations needs to be used by multiple rules
+
+// a collection of locations where a JSON Schema object can be *used*.
+//
+// note that this does not include "components.schemas" to avoid duplication.
+// this collection should be used in a rule that has "resolved" set to "true".
+// we separately validate that all schemas in "components" need to be used.
+const schemas = [
+  '$.paths[*][parameters][*].schema',
+  '$.paths[*][parameters][*].content[*].schema',
+  '$.paths[*][*][parameters][*].schema',
+  '$.paths[*][*][parameters,responses][*].content[*].schema',
+  '$.paths[*][*][requestBody].content[*].schema'
```

Comment:
Don't these all need recursion to account for schemas referenced/embedded in object schema properties?

Reply:
Addressed in the functions - let's use the thread off your other comment to discuss

Reply:
Well, I was thinking along the lines of this (for request body schemas, as an example):

```
'$.paths[*][*][requestBody].content[*].schema',
'$.paths[*][*][requestBody].content[*].schema..properties[*]',
'$.paths[*][*][requestBody].content[*].schema..items'
```

But now that I look closely, I think that could result in false positives for object schema properties named `properties` or `items` or for the use of those key names within extensions. So... maybe there is no way around the manual recursion. :-(